xtra/src/message_channel.rs

```diff
@@ -366,4 +380,64 @@ where
     {
         Box::new(Address(self.0.to_tx_either()))
     }
+
+    fn hash(&self, state: &mut dyn Hasher) {
+        state.write_usize(self.0.inner_ptr() as *const _ as usize);
+        state.write_u8(self.0.is_strong() as u8);
+        state.finish();
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use std::collections::HashSet;
+
+    use crate::{Actor, Handler, Mailbox};
+
+    type TestMessageChannel = super::MessageChannel<TestMessage, ()>;
+
+    #[derive(Hash, PartialEq)]
+    struct ChannelKey(TestMessageChannel);
+    impl Eq for ChannelKey {}
+
+    struct TestActor;
+    struct TestMessage;
+
+    impl Actor for TestActor {
+        type Stop = ();
+
+        async fn stopped(self) -> Self::Stop {}
+    }
+
+    impl Handler<TestMessage> for TestActor {
+        type Return = ();
+
+        async fn handle(&mut self, _: TestMessage, _: &mut crate::Context<Self>) -> Self::Return {}
+    }
+
+    #[test]
+    fn hashcode_and_partial_eq() {
```

Comment:
This only tests hash-code, right?

Would you mind adding one for `PartialEq` too please? 🙏 (I do realise that we should have had one for that from the beginning.)

Reply:
Hashcodes are permitted to collide (they only guarantee inequality), so HashSet will call PartialEq (via Eq) during the second call to insert.

Reply:
Hmm, I should also check the negative case, making the change now.

Reply:
I have completely changed the approach and avoid using `HashSet` as a property test. We know that the source data for the hashcode should be different, so that is tested directly.